ccl/backupccl: TestOnlineRestoreBasic failed

ccl/backupccl.TestOnlineRestoreBasic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16266828?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16266828?buildTab=artifacts#/) on master @ [e8e4cce43547c18c7cb988c42c3f2b7065717f51](https://github.com/cockroachdb/cockroach/commits/e8e4cce43547c18c7cb988c42c3f2b7065717f51):


```
=== RUN   TestOnlineRestoreBasic
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic2773444410
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:299: no Invalid Descriptors
    testutils.go:299: no Invalid Descriptors
    restore_online_test.go:96: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/44cc1cc0ef77f902d6c266f2a388abb1/logTestOnlineRestoreBasic2773444410
--- FAIL: TestOnlineRestoreBasic (5.74s)
=== RUN   TestOnlineRestoreBasic/incremental=true
    restore_online_test.go:90: query 'SELECT create_statement FROM [SHOW CREATE TABLE data.bank]': expected:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        )
        
        got:
        CREATE TABLE public.bank (
        	id INT8 NOT NULL,
        	balance INT8 NULL,
        	payload STRING NULL,
        	CONSTRAINT bank_pkey PRIMARY KEY (id ASC),
        	FAMILY fam_0_id_balance_payload (id, balance, payload)
        ) WITH (sql_stats_automatic_collection_enabled = false)
        
    --- FAIL: TestOnlineRestoreBasic/incremental=true (1.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreBasic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40671

--- a/pkg/ccl/backupccl/restore_online_test.go
+++ b/pkg/ccl/backupccl/restore_online_test.go
@@ -82,10 +82,7 @@ func TestOnlineRestoreBasic(t *testing.T) {
 		assertMVCCOnlineRestore(t, rSQLDB, preRestoreTs)
 		assertOnlineRestoreWithRekeying(t, sqlDB, rSQLDB)
 
-		// Wait for the download job to complete.
-		var downloadJobID jobspb.JobID
-		rSQLDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%'`).Scan(&downloadJobID)
-		jobutils.WaitForJobToSucceed(t, rSQLDB, downloadJobID)
+		waitForLatestDownloadJobToSucceed(t, rSQLDB)
 
 		rSQLDB.CheckQueryResults(t, createStmt, createStmtRes)
 		sqlDB.CheckQueryResults(t, jobutils.GetExternalBytesForConnectedTenant, [][]string{{"0"}})
@@ -107,12 +104,12 @@ func TestOnlineRestorePartitioned(t *testing.T) {
 	)
 	defer cleanupFn()
 
-	sqlDB.Exec(t, `BACKUP DATABASE data TO ('nodelocal://1/a?COCKROACH_LOCALITY=default', 
-		'nodelocal://1/b?COCKROACH_LOCALITY=dc%3Ddc2', 
+	sqlDB.Exec(t, `BACKUP DATABASE data TO ('nodelocal://1/a?COCKROACH_LOCALITY=default',
+		'nodelocal://1/b?COCKROACH_LOCALITY=dc%3Ddc2',
 		'nodelocal://1/c?COCKROACH_LOCALITY=dc%3Ddc3')`)
 
-	j := sqlDB.QueryStr(t, `RESTORE DATABASE data FROM ('nodelocal://1/a?COCKROACH_LOCALITY=default', 
-		'nodelocal://1/b?COCKROACH_LOCALITY=dc%3Ddc2', 
+	j := sqlDB.QueryStr(t, `RESTORE DATABASE data FROM ('nodelocal://1/a?COCKROACH_LOCALITY=default',
+		'nodelocal://1/b?COCKROACH_LOCALITY=dc%3Ddc2',
 		'nodelocal://1/c?COCKROACH_LOCALITY=dc%3Ddc3') WITH new_db_name='d2', EXPERIMENTAL DEFERRED COPY`)
 
 	srv.Servers[0].JobRegistry().(*jobs.Registry).TestingNudgeAdoptionQueue()
@@ -209,14 +206,17 @@ func TestOnlineRestoreWaitForDownload(t *testing.T) {
 	sqlDB.Exec(t, fmt.Sprintf("BACKUP INTO '%s'", externalStorage))
 
 	sqlDB.Exec(t, fmt.Sprintf("RESTORE DATABASE data FROM LATEST IN '%s' WITH EXPERIMENTAL DEFERRED COPY, new_db_name=data2", externalStorage))
-
-	// Wait for the download job to complete.
-	var downloadJobID jobspb.JobID
-	sqlDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%' ORDER BY created DESC LIMIT 1`).Scan(&downloadJobID)
-	jobutils.WaitForJobToSucceed(t, sqlDB, downloadJobID)
-
+	waitForLatestDownloadJobToSucceed(t, sqlDB)
 	sqlDB.CheckQueryResults(t, jobutils.GetExternalBytesForConnectedTenant, [][]string{{"0"}})
 
+}
+
+func waitForLatestDownloadJobToSucceed(t *testing.T, sqlDB *sqlutils.SQLRunner) {
+	var downloadJobID jobspb.JobID
+	sqlDB.QueryRow(t,
+		`SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%' ORDER BY created DESC LIMIT 1`,
+	).Scan(&downloadJobID)
+	jobutils.WaitForJobToSucceed(t, sqlDB, downloadJobID)
 }
 
 // TestOnlineRestoreTenant runs an online restore of a tenant and ensures the

